can/interfaces/slcan.py

```diff
@@ -244,8 +244,10 @@ def shutdown(self):
         self.serialPortOrig.close()
 
     def fileno(self):
-        if hasattr(self.serialPortOrig, "fileno"):
+        try:
             return self.serialPortOrig.fileno()
+        except:
```

Comment:
@bessman is right, we could for example use
```suggestion
        except (NotImplementedError, AttributeError):
```

Reply:
The issue is that on windows fileno exists.
the pyserial code raises an exception.
the serial object inherits from io.IOBase which defines a fileno method.
according to the docs (https://docs.python.org/3/library/io.html#io.IOBase.fileno) it can raise an OSError. In practice in CPython io.UnsupportedOperation (https://docs.python.org/3/library/io.html#io.UnsupportedOperation) is raised.

Which of these should we catch?

Reply:
In general, you want to be as specific as possible when catching exceptions. `except OSError:` would catch UnsupportedOperation, but it would also catch any other error which inherits from OSError, which we do not intend. Therefore, `except UnsupportedOperation:` is preferable.

Reply:
@bessman the issue i have with catching UnsupportedOperation is that according to the documentation it could be any derivative of OSError.

Reply:
I think UnsupportedOperation sound good.